### PR TITLE
Dont prune ucrs

### PR DESCRIPTION
**Pull request by emord**
# uses parts from #12585 and #12992 so wait on those until merge (should only be one commit after those)

### CODE REVIEW

**Comment by emord** [comment]
buddy @czue

File: corehq/apps/userreports/sql/__init__.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 from .adapter import get_indicator_table, IndicatorSqlAdapter, metadata, ErrorRaisingIndicatorSqlAdapter
 from .columns import get_expanded_column_config, SqlColumnConfig
-from .util import get_column_name, get_table_name, truncate_value
+from .util import get_column_name
```
Comment (dimagimon) [inline corehq/apps/userreports/sql/__init__.py]: F401 'get_column_name' imported but unused

**Comment by czue** [comment]
wondering how scalable this will be if we have thousands+ of UCRs defined. at that point maybe even just getting a list of indexes from ES would be problematic(?). but it might be cleaner if we could do this with a prefix or something that didn't load all UCR docs from the db

**Comment by czue** [comment]
(that comment not blocking, just noting as something we might need to change in the future)

File: corehq/apps/userreports/dbaccessors.py [hunk 2]
```diff
@@ -50,3 +50,16 @@ def get_all_report_configs():
...
     delete_all_docs_by_doc_type(ReportConfiguration.get_db(), ('ReportConfiguration',))
+
+
+def _get_all_data_sources():
+    from corehq.apps.userreports.models import DataSourceConfiguration
+    return DataSourceConfiguration.view(
+        'userreports/data_sources_by_build_info',
+        reduce=False,
+        include_docs=True
+    )
```
Comment (czue) [inline corehq/apps/userreports/dbaccessors.py]: nit: maybe use `iter_docs` for this?

Comment (czue) [inline corehq/apps/userreports/dbaccessors.py]: we could even.... add a couch view to just pull the index names. :smiling_imp: 

it is just the meta db so shouldn't be expensive

**Comment by emord** [comment]
> but it might be cleaner if we could do this with a prefix on the index name

It does have a prefix: https://github.com/dimagi/commcare-hq/blob/dont-prune-ucrs/corehq/apps/userreports/util.py#L133

I was thinking that eventually we would want to prune ucrs from lapsed subscriptions or from those that aren't currently referenced in a report. I suppose that could be a separate management command though. Will revisit once we decide that UCR ES is the way to go

**Comment by czue** [comment]
sounds good. 
:+1: on :white_check_mark:

**Pull request merged**